### PR TITLE
Added todo feature in the app

### PR DESCRIPTION
**Pull request by shubhmore1212**

### CODE REVIEW

File: src/Pages/ShowTask.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const ShowTask = (props) => {
+  const tasks = props.tasks;
+
+  return (
+    <>
```
Comment (vaibhavchalse99) [inline src/Pages/ShowTask.jsx:5]: do we really need this <></> ?

File: src/Pages/ShowTask.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      <div className="show-task">
+        {tasks.map((task, index) => {
```
Comment (vaibhavchalse99) [inline src/Pages/ShowTask.jsx:7]: use 
{tasks.map((task, index) => (
            <div className="show-task-map" key={index}>
              <input
                className="show-task-input"
                type="checkbox"
                onChange={() => props.statusHandler(task.id, index)}
              />
              <label className="strikethrough">{task.body}</label>
            </div>
          ))}

File: src/Pages/ShowTask.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+              <label className="strikethrough">{task.body}</label>
+            </div>
+          );
+        })}
+      </div>
+    </>
+  );
+};
+
+export default ShowTask;
```
Comment (vaibhavchalse99) [inline src/Pages/ShowTask.jsx:24]: wrap it inside the React.memo()
find out the reason why we do this ?

File: src/App.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import { useState } from "react";
```
Comment (vaibhavchalse99) [inline src/App.js:1]: here  onwards we should follow the import sequence as follows
1 external librares
2 custom components 
3 helper files
4 constant files
5 media, css files

there should be blank line in between this sections

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      id: 3,
+      status: "pending",
+      body: "Write Assignments",
+    },
+  ]);
+
+  const [newTask, setNewTask] = useState("");
+
+  const newTaskHandler = (e) => {
+    console.log(e.target.value);
```
Comment (vaibhavchalse99) [inline src/App.js:26]: remove log 
code should go without logs

File: src/App.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    setNewTask(e.target.value);
+  };
+
+  const addTask = (e) => {
+    let tempObj = {
+      id: tasks[tasks.length - 1].id + 1,
+      status: "pending",
+      body: newTask,
+    };
+    console.log(tempObj, tasks);
```
Comment (vaibhavchalse99) [inline src/App.js:37]: remove this log

File: src/App.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+  const newTaskHandler = (e) => {
+    console.log(e.target.value);
+
+    setNewTask(e.target.value);
+  };
+
+  const addTask = (e) => {
+    let tempObj = {
+      id: tasks[tasks.length - 1].id + 1,
```
Comment (vaibhavchalse99) [inline src/App.js:33]: can wer use task.length for Id ?

File: src/App.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      status: "pending",
+      body: newTask,
+    };
+    console.log(tempObj, tasks);
+    tasks.push(tempObj);
+    const tempTasks = tasks;
+
+    setTasks([...tasks]);
```
Comment (vaibhavchalse99) [inline src/App.js:41]: we don't do like this 
use 
    setTasks([...tasks, tempObj]);

File: src/App.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    setNewTask("");
+  };
+
+  const statusHandler = (id, index) => {
+    let doneTask = tasks.filter((task) => task.id === id);
```
Comment (vaibhavchalse99) [inline src/App.js:46]: use map.

File: src/App.js (hunk 4 above)
Comment (shubhmore1212) [reply src/App.js:1]: Done

File: src/App.js (hunk 5 above)
Comment (shubhmore1212) [reply src/App.js:26]: Done

File: src/App.js (hunk 6 above)
Comment (shubhmore1212) [reply src/App.js:37]: Done

File: src/App.js (hunk 8 above)
Comment (shubhmore1212) [reply src/App.js:41]: Done

File: src/App.js (hunk 9 above)
Comment (shubhmore1212) [reply src/App.js:46]: Done

File: src/App.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    } else {
+      currentStatus = "pending";
+    }
+
+    tasks[index].status = currentStatus;
+    setTasks([...tasks]);
+  };
+
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (shubhmore1212) [reply src/App.js:60]: Yes, if there is a div then no need of this <> ( i.e. React.Fragment)

File: src/Pages/ShowTask.jsx [hunk 11]
```diff
@@ -0,0 +1,24 @@
+const ShowTask = (props) => {
+  const tasks = props.tasks;
```
[the remark this answers is not in the document]
Comment (shubhmore1212) [reply src/Pages/ShowTask.jsx:2]: Done

File: src/Pages/ShowTask.jsx (hunk 1 above)
Comment (shubhmore1212) [reply src/Pages/ShowTask.jsx:5]: No need here as i have used div.

File: src/Pages/ShowTask.jsx (hunk 2 above)
Comment (shubhmore1212) [reply src/Pages/ShowTask.jsx:7]: Done

**Review by vaibhavchalse99 — APPROVED** [review]

**Pull request merged**